gradio/components/chatbot.py

```diff
@@ -139,24 +151,37 @@ def __init__(
         self.placeholder = placeholder
 
     def _preprocess_chat_messages(
-        self, chat_message: str | FileMessage | None
-    ) -> str | tuple[str | None] | tuple[str | None, str] | None:
+        self,
+        chat_message: str
+        | FileMessage
+        | List[FileMessage]
+        | GalleryData
+        | PlotData
+        | None,
+    ) -> str | GradioComponent | tuple[str | None] | tuple[str | None, str] | None:
         if chat_message is None:
             return None
         elif isinstance(chat_message, FileMessage):
             if chat_message.alt_text is not None:
                 return (chat_message.file.path, chat_message.alt_text)
             else:
                 return (chat_message.file.path,)
-        elif isinstance(chat_message, str):
+        elif isinstance(chat_message, (str)):
             return chat_message
+        elif isinstance(chat_message, GalleryData):
```

Comment:
I think if a user returns `gr.Image` in the conversation, a `gr.Image` should be present in the history. That's not the case now, but it should be possible if we implement one of the api suggestions above.